database/sqlite/sqlite_aclk_alert.c

```diff
@@ -1047,55 +1048,40 @@ void aclk_push_alarm_checkpoint(RRDHOST *host __maybe_unused)
         return;
     }
 
-    //TODO: make sure all pending events are sent.
     if (rrdhost_flag_check(host, RRDHOST_FLAG_ACLK_STREAM_ALERTS)) {
         //postpone checkpoint send
         wc->alert_checkpoint_req++;
         log_access("ACLK REQ [%s (N/A)]: ALERTS CHECKPOINT POSTPONED", rrdhost_hostname(host));
         return;
     }
 
-    //TODO: lock rc here, or make sure it's called when health decides
-    //count them
     RRDCALC *rc;
     uint32_t cnt = 0;
     size_t len = 0;
-    active_alerts_t *active_alerts = NULL;
 
+    active_alerts_t *active_alerts = callocz(BATCH_ALLOCATED, sizeof(active_alerts_t));
     foreach_rrdcalc_in_rrdhost_read(host, rc) {
         if(unlikely(!rc->rrdset || !rc->rrdset->last_collected_time.tv_sec))
             continue;
 
         if (rc->status == RRDCALC_STATUS_WARNING ||
             rc->status == RRDCALC_STATUS_CRITICAL) {
 
+            if (cnt && !(cnt % BATCH_ALLOCATED)) {
```

Comment:
I think this can be changed to 
```
if (!(cnt % BATCH_ALLOCATED)) {
```
if the above line is changed back to:
```
active_alerts_t *active_alerts = NULL;
```